### PR TITLE
fix: Support for PCRetriableException in ReactorProcessor (#733)

### PR DESCRIPTION
**Pull request by brp-jakob**
* Support for PCRetriableException in ReactorProcessor.
* Error handling moved to subscribe param to avoid onErrorDropped.

### Checklist

- [ ] Documentation (if applicable)
- [x] Changelog

### CODE REVIEW

File: parallel-consumer-reactor/src/main/java/io/confluent/parallelconsumer/reactor/ReactorProcessor.java [hunk 1]
```diff
@@ -102,26 +103,10 @@ public void react(Function<PollContext<K, V>, Publisher<?>> reactorFunction) {
...
-                        pollContext.streamWorkContainers().forEach(wc -> {
-                            wc.onUserFunctionFailure(throwable);
-                            addToMailbox(pollContext, wc);
-                        });
-                    })
+                    .doOnNext(signal -> log.trace("doOnNext {}", signal))
                     // cause users Publisher to run a thread pool, if it hasn't already - this is a crucial magical part
                     .subscribeOn(getScheduler())
-                    .subscribe();
+                    .subscribe(ignore -> onComplete(pollContext), throwable -> onError(pollContext, throwable));
```
Comment (rkolesnev) [inline parallel-consumer-reactor/src/main/java/io/confluent/parallelconsumer/reactor/ReactorProcessor.java:109]: Hey @brp-jakob - can you please explain why the change from `doOnComplete(successCallback)` / `doOnError(errorCallback)` is needed to `subscribe(successCallback, errorCallback)` ?

Comment (brp-jakob) [reply parallel-consumer-reactor/src/main/java/io/confluent/parallelconsumer/reactor/ReactorProcessor.java:109]: Its this part:
"Error handling moved to subscribe param to avoid onErrorDropped."

There is a long discussion about this here (and in related issues to it): https://github.com/reactor/reactor-core/issues/1431

**Comment by rkolesnev** [comment]
@brp-jakob - could you please regenerate the Readme - either by using maven goal "README" or running `mvn asciidoc-template:build`? That will re-generate the readme incorporating the updated Changelog from this PR.

Ignore any errors running it - the asciidoc plugin works recursively on all sub-modules but only top level project actually has readme and the template so it logs errors not finding template on sub-modules but that is a false error.

**Review by rkolesnev — APPROVED** [review]
